### ISSUE TITLE
Duplicates all docker containers on reboot

### ISSUE THREAD

**Issue by carldanley** [kubernetes/kubernetes#37507]
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.):
No.

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):
"duplicate", "reboot", "restart"

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
bug report

**Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.4", GitCommit:"3b417cc4ccd1b8f38ff9ec96bb50a81ca0ea9d56", GitTreeState:"clean", BuildDate:"2016-10-21T02:48:38Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.4", GitCommit:"3b417cc4ccd1b8f38ff9ec96bb50a81ca0ea9d56", GitTreeState:"clean", BuildDate:"2016-10-21T02:42:39Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: Hardware configuration
- **OS** (e.g. from /etc/os-release): Ubuntu 16.04
- **Kernel** (e.g. `uname -a`): `Linux radium-control 4.4.0-47-generic #68-Ubuntu SMP Wed Oct 26 19:39:52 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux`
- **Install tools**: `kubeadm`
- **Others**:

**What happened**:
On a fresh install, I rebooted all the nodes and instead of the pre-existing ones being re-used, it re-created all new ones and left the old ones there.

```
CONTAINER ID        IMAGE                                                           COMMAND                  CREATED             STATUS                      PORTS                    NAMES
ffede927ca21        gcr.io/google_containers/kube-discovery-amd64:1.0               "/usr/local/bin/kube-"   22 minutes ago      Up 22 minutes                                        k8s_kube-discovery.95f6cda5_kube-discovery-1150918428-2fuii_kube-system_de8d448f-b397-11e6-ad3d-001b78777574_9ef8ccf2
b633ebb5bdeb        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 22 minutes ago      Up 22 minutes                                        k8s_POD.d8dbe16c_kube-discovery-1150918428-2fuii_kube-system_de8d448f-b397-11e6-ad3d-001b78777574_c8f5e24e
04ff23f8f1f8        weaveworks/weave-npc:1.8.1                                      "/usr/bin/weave-npc"     22 minutes ago      Up 22 minutes                                        k8s_weave-npc.58d9497_weave-net-ge857_kube-system_f5251ff3-b392-11e6-a9de-001b78777574_c6682817
7f9f4f49a4d8        weaveworks/weave-kube:1.8.1                                     "/home/weave/launch.s"   22 minutes ago      Up 22 minutes                                        k8s_weave.92f5ac48_weave-net-ge857_kube-system_f5251ff3-b392-11e6-a9de-001b78777574_d4c483f4
f217ed69e66b        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 22 minutes ago      Up 22 minutes                                        k8s_POD.d8dbe16c_weave-net-ge857_kube-system_f5251ff3-b392-11e6-a9de-001b78777574_cdcd9e35
7475149bd570        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 22 minutes ago      Up 22 minutes                                        k8s_dummy.3ba1b90_dummy-2088944543-scso2_kube-system_48c16d8c-b392-11e6-a9de-001b78777574_7d76b048
19fe847464ec        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 22 minutes ago      Up 22 minutes                                        k8s_POD.d8dbe16c_dummy-2088944543-scso2_kube-system_48c16d8c-b392-11e6-a9de-001b78777574_615c31b5
fcd05cfaa1f8        gcr.io/google_containers/kube-proxy-amd64:v1.4.4                "kube-proxy --v=2 --k"   22 minutes ago      Up 22 minutes                                        k8s_kube-proxy.773bb601_kube-proxy-ucdmq_kube-system_4a7045e5-b392-11e6-a9de-001b78777574_a781a8f6
7b82f47e1aa8        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 22 minutes ago      Up 22 minutes                                        k8s_POD.d8dbe16c_kube-proxy-ucdmq_kube-system_4a7045e5-b392-11e6-a9de-001b78777574_78c77c8f
3fc4beb8f823        gcr.io/google_containers/kube-controller-manager-amd64:v1.4.4   "kube-controller-mana"   22 minutes ago      Up 22 minutes                                        k8s_kube-controller-manager.edfea2a0_kube-controller-manager-radium-control_kube-system_aa642bd88d7c4f8636a7f3d6df731565_9d864923
deca928f2d8d        gcr.io/google_containers/etcd-amd64:2.2.5                       "etcd --listen-client"   22 minutes ago      Up 22 minutes                                        k8s_etcd.5a0e984b_etcd-radium-control_kube-system_a81a848a0bf7f0d38bc5f9b6b3a7140a_7f48caf6
77a4c87502ec        gcr.io/google_containers/kube-apiserver-amd64:v1.4.4            "kube-apiserver --v=2"   22 minutes ago      Up 22 minutes                                        k8s_kube-apiserver.bfb4d438_kube-apiserver-radium-control_kube-system_6bd1df9a370394cb83058fae4f9381dd_6670a8f1
bb68da396500        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 22 minutes ago      Up 22 minutes                                        k8s_POD.d8dbe16c_kube-controller-manager-radium-control_kube-system_aa642bd88d7c4f8636a7f3d6df731565_8c94f7a0
0beae50eb6e7        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 22 minutes ago      Up 22 minutes                                        k8s_POD.d8dbe16c_etcd-radium-control_kube-system_a81a848a0bf7f0d38bc5f9b6b3a7140a_e581ece1
185aafc4201c        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 22 minutes ago      Up 22 minutes                                        k8s_POD.d8dbe16c_kube-apiserver-radium-control_kube-system_6bd1df9a370394cb83058fae4f9381dd_f6b24953
fe1b90d781e6        gcr.io/google_containers/kube-scheduler-amd64:v1.4.4            "kube-scheduler --v=2"   22 minutes ago      Up 22 minutes                                        k8s_kube-scheduler.c164d8b1_kube-scheduler-radium-control_kube-system_87687297b5040798a1737ed4061b3dab_cd1a4525
0656dae45c34        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 22 minutes ago      Up 22 minutes                                        k8s_POD.d8dbe16c_kube-scheduler-radium-control_kube-system_87687297b5040798a1737ed4061b3dab_7fc6d0cb
010fca2aa9ea        weaveworks/weave-npc:1.8.1                                      "/usr/bin/weave-npc"     56 minutes ago      Exited (0) 24 minutes ago                            k8s_weave-npc.58d9497_weave-net-ge857_kube-system_f5251ff3-b392-11e6-a9de-001b78777574_45656cd2
af45cb766550        weaveworks/weave-kube:1.8.1                                     "/home/weave/launch.s"   57 minutes ago      Exited (0) 24 minutes ago                            k8s_weave.92f5ac48_weave-net-ge857_kube-system_f5251ff3-b392-11e6-a9de-001b78777574_b627f7a9
dd864b73893b        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 57 minutes ago      Exited (0) 24 minutes ago                            k8s_POD.d8dbe16c_weave-net-ge857_kube-system_f5251ff3-b392-11e6-a9de-001b78777574_d18d6963
c58758f61ece        gcr.io/google_containers/kube-proxy-amd64:v1.4.4                "kube-proxy --v=2 --k"   About an hour ago   Exited (0) 24 minutes ago                            k8s_kube-proxy.773bb601_kube-proxy-ucdmq_kube-system_4a7045e5-b392-11e6-a9de-001b78777574_c0672a94
d8f03c785fcd        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 About an hour ago   Exited (0) 24 minutes ago                            k8s_POD.d8dbe16c_kube-proxy-ucdmq_kube-system_4a7045e5-b392-11e6-a9de-001b78777574_d6c09a7a
730300be084c        gcr.io/google_containers/kube-discovery-amd64:1.0               "/usr/local/bin/kube-"   About an hour ago   Exited (0) 24 minutes ago                            k8s_kube-discovery.95f6cda5_kube-discovery-1150918428-y43bp_kube-system_49d67e85-b392-11e6-a9de-001b78777574_e86c70ed
c7787b36ac00        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 About an hour ago   Exited (0) 24 minutes ago                            k8s_POD.d8dbe16c_kube-discovery-1150918428-y43bp_kube-system_49d67e85-b392-11e6-a9de-001b78777574_bc686f73
716eb3f55c48        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 About an hour ago   Exited (0) 24 minutes ago                            k8s_dummy.3ba1b90_dummy-2088944543-scso2_kube-system_48c16d8c-b392-11e6-a9de-001b78777574_13b2ea07
fbf59341b61c        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 About an hour ago   Exited (0) 24 minutes ago                            k8s_POD.d8dbe16c_dummy-2088944543-scso2_kube-system_48c16d8c-b392-11e6-a9de-001b78777574_6de04743
faf34c5a8871        gcr.io/google_containers/kube-controller-manager-amd64:v1.4.4   "kube-controller-mana"   About an hour ago   Exited (0) 24 minutes ago                            k8s_kube-controller-manager.edfea2a0_kube-controller-manager-radium-control_kube-system_aa642bd88d7c4f8636a7f3d6df731565_d6f31f7e
dc662a196d44        gcr.io/google_containers/kube-apiserver-amd64:v1.4.4            "kube-apiserver --v=2"   About an hour ago   Exited (0) 24 minutes ago                            k8s_kube-apiserver.bfb4d438_kube-apiserver-radium-control_kube-system_6bd1df9a370394cb83058fae4f9381dd_0eb00d1c
16c18aa7d375        gcr.io/google_containers/etcd-amd64:2.2.5                       "etcd --listen-client"   About an hour ago   Exited (0) 24 minutes ago                            k8s_etcd.5a0e984b_etcd-radium-control_kube-system_a81a848a0bf7f0d38bc5f9b6b3a7140a_4a2fc553
362bd205e080        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 About an hour ago   Exited (0) 24 minutes ago                            k8s_POD.d8dbe16c_kube-controller-manager-radium-control_kube-system_aa642bd88d7c4f8636a7f3d6df731565_5e72e742
0d7cec328920        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 About an hour ago   Exited (0) 24 minutes ago                            k8s_POD.d8dbe16c_kube-apiserver-radium-control_kube-system_6bd1df9a370394cb83058fae4f9381dd_d774ffdf
6b6ab724f832        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 About an hour ago   Exited (0) 24 minutes ago                            k8s_POD.d8dbe16c_etcd-radium-control_kube-system_a81a848a0bf7f0d38bc5f9b6b3a7140a_f4a4fe87
f66e44480edb        gcr.io/google_containers/kube-scheduler-amd64:v1.4.4            "kube-scheduler --v=2"   About an hour ago   Exited (0) 24 minutes ago                            k8s_kube-scheduler.c164d8b1_kube-scheduler-radium-control_kube-system_87687297b5040798a1737ed4061b3dab_81b91d53
99671aaf59c9        gcr.io/google_containers/pause-amd64:3.0                        "/pause"                 About an hour ago   Exited (0) 24 minutes ago                            k8s_POD.d8dbe16c_kube-scheduler-radium-control_kube-system_87687297b5040798a1737ed4061b3dab_0398d187
```

**What you expected to happen**:
I expected kubernetes to re-use the pods or at least clean up after itself.

**How to reproduce it** (as minimally and precisely as possible):
* Fresh install following http://kubernetes.io/docs/getting-started-guides/kubeadm/.
* Install weave net
* reboot server

**Anything else do we need to know**:

**Comment by huangjiasingle** [comment]
@carldanley  please look at the container's status?

**Comment by carldanley** [comment]
The "container status" you were telling me to look at is what I posted above: `Exited (0)`. Upon inspecting the logs for various containers, I found that most of them have something in their stdout that says "exited: terminated" or some other similar saying (which makes sense because in my ticket I mentioned that I restarted). This is normal docker behavior.... What I don't understand is why kubernetes didn't re-use the containers it originally created? If it needs to re-create new containers, fine; why didn't it destroy the old ones and clean up after itself.
